packages/expo-camera/src/CameraModule/UserMediaManager.ts

```diff
@@ -0,0 +1,149 @@
+/* eslint-env browser */
+import { canUseDOM } from 'fbjs/lib/ExecutionEnvironment';
+
+export let userMediaRequested: boolean = false;
+export let mountedInstances: any[] = [];
+
+async function requestLegacyUserMediaAsync(props): Promise<any[]> {
+  const optionalSource = id => ({ optional: [{ sourceId: id }] });
+
+  const constraintToSourceId = constraint => {
+    const { deviceId } = constraint;
+
+    if (typeof deviceId === "string") {
+      return deviceId;
+    }
+
+    if (Array.isArray(deviceId) && deviceId.length > 0) {
+      return deviceId[0];
+    }
+
+    if (typeof deviceId === "object" && deviceId.ideal) {
+      return deviceId.ideal;
+    }
+
+    return null;
+  };
+
+  // @ts-ignore: https://caniuse.com/#search=getSources Chrome for Android (78) & Samsung Internet (10.1) use this
+  const sources: any[] = await (new Promise((resolve) => MediaStreamTrack.getSources(sources => resolve(sources))))
+
+  let audioSource = null;
+  let videoSource = null;
+
+  sources.forEach(source => {
+    if (source.kind === "audio") {
+      audioSource = source.id;
+    } else if (source.kind === "video") {
+      videoSource = source.id;
+    }
+  });
+
+  const audioSourceId = constraintToSourceId(props.audioConstraints);
+  if (audioSourceId) {
+    audioSource = audioSourceId;
+  }
+
+  const videoSourceId = constraintToSourceId(props.videoConstraints);
+  if (videoSourceId) {
+    videoSource = videoSourceId;
+  }
+
+  return [
+    optionalSource(audioSource),
+    optionalSource(videoSource)
+  ]
+}
+
+async function sourceSelectedAsync(isMuted: boolean, audioConstraints?: MediaTrackConstraints | boolean, videoConstraints?: MediaTrackConstraints | boolean): Promise<MediaStream> {
+  const constraints: MediaStreamConstraints = {
+    video: typeof videoConstraints !== "undefined" ? videoConstraints : true
```

Comment:
I suspect you do it intentionally (`typeof videoConstraints !== "undefined"`) ?
```suggestion
    video: !!videoConstraints
```